internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -803,8 +819,66 @@ func (t *Translator) buildCircuitBreaker(policy *egv1a1.BackendTrafficPolicy) *i
 	return cb
 }
 
-func setCircuitBreakerPolicyErrorCondition(policy *egv1a1.BackendTrafficPolicy, errMsg string) {
-	message := fmt.Sprintf("Unable to translate Circuit Breaker: %s", errMsg)
+func (t *Translator) buildTimeout(policy *egv1a1.BackendTrafficPolicy, r *ir.HTTPRoute) *ir.Timeout {
+	var tto *ir.TCPTimeout
+	var hto *ir.HTTPTimeout
+
+	pto := policy.Spec.Timeout
+
+	if pto.TCP != nil && pto.TCP.ConnectTimeout != nil {
+		d, err := time.ParseDuration(string(*pto.TCP.ConnectTimeout))
+		if err != nil {
+			setBackendTrafficPolicyTranslationErrorCondition(policy, "TCP Timeout", fmt.Sprintf("invalid ConnectTimeout value %s", *pto.TCP.ConnectTimeout))
+			return nil
```

Comment:
we should log error in status and continue processing instead of returning early here